vendor/github.com/openshift/cluster-api/pkg/controller/machine/controller.go

```diff
@@ -128,6 +199,20 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	name := m.Name
 	klog.Infof("Reconciling Machine %q", name)
 
+	// Set/unset failing labels
+	changed, err := r.checkStatusAndLabels(m)
```

Comment:
Maybe instead of doing this at the start of the loop, we can use it as an error handler.  For instance, if create is returning 401 unauthorized, we can capture that info somehow (as a custom error or some other mechanism) and if we detect that during the create-labels logic above, we can set a more-specific error label such as 'cloud-unauthorized' or similar.